src/fielddef.ts

```diff
@@ -250,6 +252,13 @@ export function defaultType(fieldDef: FieldDef, channel: Channel): Type {
 export function normalize(fieldDef: ChannelDef, channel: Channel) {
   // If a fieldDef contains a field, we need type.
   if (isFieldDef(fieldDef)) { // TODO: or datum
+    if (fieldDef.bin) {
+      fieldDef.bin = isBoolean(fieldDef.bin) ? {maxbins: 10} : fieldDef.bin;
```

Comment:
1) As I have commented twice ([1](https://github.com/vega/vega-lite/pull/1983#discussion_r110508590) and [2](https://github.com/vega/vega-lite/pull/1983#discussion_r110524611)), please don't cause side effect!

2) Why don't you bring back `autoMaxBins` and call it here?  Now we got channel in the context and thus we don't really have to remove it. 

3) If you have a convincing reason why you should not bring  `autoMaxBins` back, at least use the default `maxbins` config here.  (If you use the old `autoMaxBins`, just get rid of the `maxbins` config for now.